src/Raven.Client/Documents/Indexes/ExpressionStringBuilder.cs

```diff
@@ -1324,8 +1324,16 @@ protected override Expression VisitMember(MemberExpression node)
                         return node; // we don't have nullable type on the server side, we can safely ignore this.
                 }
             }
-
             var exprType = node.Expression != null ? node.Member.DeclaringType : node.Type;
+#if FEATURE_DATEONLY_TIMEONLY_SUPPORT
+            if (node.Type == typeof(TimeOnly) || node.Type == typeof(DateOnly))
+            {
+                Out($"As{(node.Type == typeof(TimeOnly) ? "TimeOnly" : "DateOnly")}(");
```

Comment:
Wouldn't it be simpler?

```suggestion
                Out($"As{node.Type.Name)}(");
```

Reply:
Sure, changed.